x/concentrated-liquidity/README.md

```diff
@@ -472,31 +472,275 @@ func (k Keeper) withdrawPosition(
+/////////////////////////////////////////////////////////  <---position by user A
+-1000           -34          0       100              1000
+```
+
+The swap state is initialized as follows:
+
+- `amountSpecifiedRemaining` is set to 5_000 tokens one in specified by the user.
+- `amountCalculated` is set to zero.
+- `sqrtPrice` is set to the current sqrt price of the pool (computed from the tick -34)
+- `tick` is set to the current tick of the pool (-34)
+- `liquidity` is set to the current liquidity tracked by the pool at tick -34 (10_000)
+- `feeGrowthGlobal` is set to (0)
+
+We proceed by getting the next initialized tick in the direction of the swap (0).
+
+Each initialized tick has 2 fields:
+
+- `liquidity_gross` - this is the total liquidity referencing that tick
+at tick -1000: 10_000
+at tick 0:      1_000
+at tick 100:    1_000
+at tick 1000:  10_000
+
+- `liquidity_net` - liquidity that needs to be added to the active liquidity as we cross the tick moving in the positive direction
+so that the active liquidity is always the sum of all `liquidity_net` amounts of initialized ticks below the current one. 
+at tick -1000: 10_000
+at tick 0:      1_000
+at tick 100:   -1_000
+at tick 1000: -10_000
+
+Next, we compute swap step from tick -34 to tick 0. Assume that 5_000 tokens one in is more
+than enough to cross tick 0 and it returns 10_000 of token zero out while consuming half of
+token one in (2500).
+
+Now, we update the swap state as follows:
+
+- `amountSpecifiedRemaining` is set to 5000 - 2_500 = 2_500 tokens one in remaining.
+
+- `amountCalculated` is set to 10_000 tokens zero out calculated.
+
+- `sqrtPrice` is set to the sqrt price of the crossed initialzed tick 0 (0).
+
+- `tick` is set to the tick of the crossed initialzed tick 0 (0).
+
+- `liquidity` is set to the new old liquidity value (10_000) + the `liquidity_net` of the crossed tick 0 (1_000) = 11_000.
+
+- `feeGrowthGlobal` is set to 0 because we assumed zero swap fee.
+
+Now, we proceed by getting the next initialized tick in the direction of the swap (100).
+
+Next, we compute swap step from tick 0 to tick 100. Assume that 2_500 remaining tokens one in
+is not enough to reach the next initialized tick 100 and it returns 12_500 of token zero out while
+only reaching tick 70. The reason why we now get a greater amount of token zero out for the same
+amount of token one in is because the liquidity in this tick range is greater than the liquidity in
+the previous tick range.
+
+Now, we update the swap state as follows:
+
+- `amountSpecifiedRemaining` is set to 2_500 - 2_500 = 0 tokens one in remaining.
+
+- `amountCalculated` is set to 10_000 + 12_500 = 22_500 tokens zero out calculated.
+
+- `sqrtPrice` is set to the reached sqrt price.
+
+- `tick` is set to an uninitialized tick associated with the reached sqrt price (70).
+
+- `liquidity` is set kept the same as we did not cross any initialized tick.
+
+- `feeGrowthGlobal` is set to 0 because we assumed zero swap fee.
```

Comment:
I think swap fee would also be valuable to add in a new section. At least describing how the swap fee gets distributed to the positions after a successful swap.

Reply:
Updated: [1bd2f8d](https://github.com/osmosis-labs/osmosis/pull/4696/commits/1bd2f8d73184de3dec1b8db63304be1d84b9e658)

Planning to also write up a better section on fees separately